File: test/fixtures/sample.swift
/**
 * Swift Sample File for Method Extraction Testing
 */

class Calculator {
    private var value: Int

    init(initialValue: Int = 0) {
        self.value = initialValue
    }

    func add(_ x: Int, _ y: Int) -> Int {
        return x + y
    }

    private func subtract(_ x: Int, _ y: Int) -> Int {
        return x - y
    }

    static func multiply(_ x: Int, _ y: Int) -> Int {
        return x * y
    }

    func asyncOperation() async -> Int {
        return await fetchData()
    }

    class func classMethod() -> String {
        return "Class method"
    }
}

func standaloneFunction(param: Int) -> Int {
    return param * 2
}

func genericFunction<T>(item: T) -> T {
    return item
}

public func publicFunction() {
    print("Public function")
}

private func privateFunction() {
    print("Private function")
}

fileprivate func filePrivateFunction() {
    print("File private function")
}

struct Point {
    var x: Double
    var y: Double

    func distance() -> Double {
        return sqrt(x * x + y * y)
    }
}

File: test/fixtures/sample.scala
/**
 * Scala Sample File for Method Extraction Testing
 */

class Calculator(private var value: Int = 0) {

  def add(x: Int, y: Int): Int = {
    x + y
  }

  private def subtract(x: Int, y: Int): Int = {
    x - y
  }

  def multiply(x: Int, y: Int): Int = x * y

  def asyncOperation(): Future[Int] = Future {
    value
  }
}

object Calculator {
  def apply(value: Int): Calculator = new Calculator(value)

  def multiply(x: Int, y: Int): Int = {
    x * y
  }
}

trait Computable {
  def compute(): Int
}

case class Point(x: Double, y: Double) {
  def distance(): Double = {
    math.sqrt(x * x + y * y)
  }
}

def standaloneFunction(param: Int): Int = {
  param * 2
}

val lambdaFunction = (x: Int) => x + 1

def higherOrderFunction(f: Int => Int): Int = {
  f(10)
}

def genericFunction[T](item: T): T = {
  item
}

override def toString: String = {
  s"Calculator($value)"
}
